app/main.py

```diff
@@ -1 +1,35 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        tp = self
+        return f"{{Name: {tp.name}, Health: {tp.health}, Hidden: {tp.hidden}}}"
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(victim: Herbivore | Carnivore) -> None:
+        if isinstance(victim, Herbivore):
+            if not victim.hidden:
```

Comment:
not sure if it is better, but maybe you can make one statement with isinstance and not victim in 1 line

Reply:
Thanks for the comments, it's really better to use that, but i don't want to combine the two `not statement`

Reply:
Yes, it's better to make the condition in one line with operator `and` instead of nested `if`